evap/staff/tools.py

```diff
@@ -82,13 +82,23 @@ def custom_redirect(url_name, *args, **kwargs):
     return HttpResponseRedirect(url + "?%s" % params)
 
 
-def delete_navbar_cache():
+def delete_navbar_cache_for_user(user):
     # delete navbar cache from base.html
+    key = make_template_fragment_key('navbar', [user.username, 'de'])
+    cache.delete(key)
+    key = make_template_fragment_key('navbar', [user.username, 'en'])
+    cache.delete(key)
+
+
+def delete_navbar_cache_for_all_users():
+    for user in UserProfile.objects.all():
+        delete_navbar_cache_for_user(user)
+
+
+def delete_navbar_cache_for_special_users():
     for user in UserProfile.objects.all():
-        key = make_template_fragment_key('navbar', [user.username, 'de'])
-        cache.delete(key)
-        key = make_template_fragment_key('navbar', [user.username, 'en'])
-        cache.delete(key)
+        if user.is_reviewer or user.is_grade_publisher:
+            delete_navbar_cache_for_user(user)
```

Comment:
following this structure, you would also need another `delete navbar_cache_*` method for the `helper_course_edit` case. how about a `delete_navbar_cache_for_users` which takes a list/queryset, and use that everywhere? the two callers of `delete_navbar_cache_for_special_users`, i don't know, maybe create a `users_that_need_their_navbar_cache_deleted_on_semester_creation_or_deletion()` method?

Reply:
...maybe `users_with_semesters_in_their_navbar`? not sure...

Reply:
where would you place this method?

Reply:
maybe let's just call the `delete_navbar_cache_for_users` method for reviewers and grade publishers there.

Reply:
yeah, that was the idea. my problem was that when calling `delete_navbar_cache_for_users`, there will be two places that need to know that these two groups need their caches cleared, so i thought about creating one function that return a list containing those two groups. but it doesn't really matter, you can also duplicate the code.

Reply:
if you create that method, i would place it where it's used.

Reply:
k, please review again.